src/crm/contollers/register.controller.ts

```diff
@@ -21,6 +22,14 @@ export class RegisterController {
   @Post()
   async create(@Body() data: CreatePersonDto): Promise<ContactListDto> {
     const contact = await this.service.createPerson(data);
-    return ContactsService.toListDto(contact);
+    try {
+      return ContactsService.toListDto(contact);
+    } catch (e) {
+      if (isEmptyObject(contact)) {
+        return new ContactListDto();
+      } else {
+        return e;
```

Comment:
Exceptions need to be thrown, not returned otherwise you will be getting 200 OK for a request that failed. I also added a comment on the client-side. there is a proper way to handle errors

Reply:
> Exceptions need to be thrown, not returned otherwise you will be getting 200 OK for a request that failed. I also added a comment on the client-side. there is a proper way to handle errors

Thanks Timo for this. The code has been amended appropriately both here and client side.